src/convert.jl

```diff
@@ -154,6 +154,32 @@ function jlvec(::Type{ZonedDateTime}, rv::RVEC, force_missing::Bool=true)
     return datetimes
 end
 
+function simplify_eltype(@nospecialize v::AbstractVector)
+    eltypes = unique(typeof.(v))
+    eltyp = first(eltypes)
+    try
+        for i in 2:length(eltypes)
+            eltyp = promote_type(eltyp, eltypes[i])
```

Comment:
No need to compute all element types beforehand:
```suggestion
    eltyp = Union{}
    try
        for x in v
            eltyp = promote_type(eltyp, typeof(x))
```

Reply:
In this case `promote_type` is called for each element of `v`, in my version it's called only for the unique eltypes. Do you think it would be faster/better?

Reply:
Only benchmarking will tell.

BTW, `eltypes = unique(typeof, v)` would avoid an allocation.